tools/tpm2_policyauthorize.c

```diff
@@ -0,0 +1,183 @@
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//**********************************************************************;
+
+#include <errno.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <limits.h>
+#include <ctype.h>
+
+#include <tss2/tss2_sys.h>
+
+#include "files.h"
+#include "log.h"
+#include "tpm2_options.h"
+#include "tpm2_policy.h"
+#include "tpm2_session.h"
+#include "tpm2_tool.h"
+#include "tpm2_util.h"
+
+typedef struct tpm2_startauthsession_ctx tpm2_startauthsession_ctx;
+struct tpm2_startauthsession_ctx {
+   const char *session_path;
+   const char *to_authorize_policy_file;
+   const char *to_authorize_policy_qualifier_file;
+   const char *verifying_public_key_name_file;
+   const char *verification_ticket_file;
+   const char *policy_out_path;
+};
+
+static tpm2_startauthsession_ctx ctx;
+
+static bool on_option(char key, char *value) {
+
+    switch (key) {
+    case 'f':
+        ctx.policy_out_path = value;
+        break;
+    case 'S':
+        ctx.session_path = value;
+    case 'a':
+        ctx.to_authorize_policy_file = value;
+        break;
+    case 'b':
+        ctx.to_authorize_policy_qualifier_file = value;
+        break;
+    case 'c':
+        ctx.verifying_public_key_name_file = value;
+        break;
+    case 'd':
+        ctx.verification_ticket_file = value;
+        break;
+    }
+    return true;
+}
+
+bool tpm2_tool_onstart(tpm2_options **opts) {
+
+    static struct option topts[] = {
+        { "policy-file",    required_argument,  NULL,   'f' },
+        { "session",        required_argument,  NULL,   'S' },
+        { "to_authorize_policy_file", required_argument, NULL, 'a' },
+        { "to_authorize_policy_qualifier_file", required_argument, NULL, 'b' },
+        { "verifying_public_key_name_file", required_argument, NULL, 'c' },
+        { "verification_ticket_file", required_argument, NULL, 'd' },
+    };
+
+    *opts = tpm2_options_new("f:S:a:b:c:d:", ARRAY_LEN(topts), topts, on_option,
+                             NULL, 0);
+
+    return *opts != NULL;
+}
+
+int tpm2_tool_onrun(TSS2_SYS_CONTEXT *sapi_context, tpm2_option_flags flags) {
+
+    UNUSED(flags);
+
+    int rc = 1;
+    tpm2_session *s = NULL;
+
+    bool fail = false;
+    if (!ctx.session_path) {
+        LOG_ERR("Must specify -S session file.");
+        fail = true;
+    }
+
+    if (!ctx.to_authorize_policy_file) {
+        LOG_ERR("Must specify -a to_authorize_policy_file name.");
+        fail = true;
+    }
+    if (!ctx.to_authorize_policy_qualifier_file) {
```

Comment:
new line between conditional blocks and next block. Perhaps refactor the "option" checks into a seperate routine called check_options() or something like that.